lib/rubocop/cop/rspec/expect_change.rb

```diff
@@ -64,9 +66,11 @@ def on_block(node)
           return unless style == :method_call
 
           expect_change_with_block(node) do |receiver, message|
-            msg = format(MSG_BLOCK, obj: receiver, attr: message)
+            obj = receiver.is_a?(RuboCop::AST::Node) ? receiver.source : receiver
```

Comment:
I have an impression that something is horribly wrong with the node pattern that I've drafted in #1037.
`receiver.is_a?(RuboCop::AST::Node)` shouldn't be needed, and I would expect `.source` to be callable on any match.

Are you up to play with the node pattern a bit more?

Please don't hesitate to ping me if you stumble some weird behaviour.

Reply:
> I have an impression that something is horribly wrong with the node pattern that I've drafted in #1037.
receiver.is_a?(RuboCop::AST::Node) shouldn't be needed, and I would expect .source to be callable on any match.

I completely agree with it, and we will study about node pattern with [this](https://docs.rubocop.org/rubocop-ast/node_pattern.html)

Reply:
We studied about node pattern and fixed the implementation.